src/libs/browser-storage.js

```diff
@@ -1,20 +1,17 @@
 import _ from 'lodash/fp'
-import { getUser } from 'src/libs/auth'
-import { maybeParseJSON } from 'src/libs/utils'
+import { maybeParseJSON, subscribable } from 'src/libs/utils'
 
 
-export const getDynamic = (storage, key) => {
-  const storageKey = `dynamic-storage/${key}`
-  const data = maybeParseJSON(storage.getItem(storageKey))
-  return data && data.value
-}
+/*
+ * This library provides a higher level interface on top of localStorage and sessionStorage.
+ * Values must be JSON-serializable. The 'dynamic' version is preferred if possible, but dynamic
+ * values might be deleted in case of space overflow. For critical data, use the 'static' version.
+ */
 
-export const setDynamic = (storage, key, value) => {
-  const storageKey = `dynamic-storage/${key}`
-  const storageValue = JSON.stringify({ timestamp: Date.now(), value })
+const forceSetItem = (storage, key, value) => {
   while (true) {
```

Comment:
Since JS has no tail call optimization, each recursive call always creates a new stack frame, which could potentially result in an overflow. I agree that this `while` in particular is tricky, because it's actually used as a retry mechanism, not for polling. There's an abstraction that we could write to extract this, but I think it's beyond the scope of this PR.